core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -151,32 +152,126 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
+              getMaximumAccessLevel(workspaceContext.workspaceId.toString)
+            } else {
+              Future.successful(WorkspaceAccessLevels.NoAccess)
+            }
+
+          DBIO.from(accessLevelFuture()) flatMap { accessLevel =>
+            // we may have calculated accessLevel because canShare/canCompute needs it;
+            // but if the user didn't ask for it, don't return it
+            val optionalAccessLevelForResponse = if (getParam(options, AccessLevel)) { Option(accessLevel) } else { None }
+
+            // determine which functions to use for the various part of the response
+            def bucketOptionsFuture: () => Future[Option[WorkspaceBucketOptions]] = () => if (getParam(options, BucketOptions)) {
+              gcsDAO.getBucketDetails(workspaceContext.workspace.bucketName, RawlsBillingProjectName(workspaceContext.workspace.namespace)).map(Option(_))
+            } else {
+              noFuture
+            }
+            def canComputeFuture: () => Future[Option[Boolean]] = () => if (getParam(options, CanCompute)) {
+              getUserComputePermissions(workspaceContext.workspaceId.toString, accessLevel).map(Option(_))
+            } else {
+              noFuture
+            }
+            def canShareFuture: () => Future[Option[Boolean]] = () => if (getParam(options, CanShare)) {
+              //convoluted but accessLevel for both params because user could at most share with their own access level
+              getUserSharePermissions(workspaceContext.workspaceId.toString, accessLevel, accessLevel).map(Option(_))
+            } else {
+              noFuture
+            }
+            def catalogFuture: () => Future[Option[Boolean]] = () => if (getParam(options, Catalog)) {
+              getUserCatalogPermissions(workspaceContext.workspaceId.toString).map(Option(_))
+            } else {
+              noFuture
+            }
+
+            def ownersFuture: () => Future[Option[Set[String]]] = () => if (options.getOrElse(WorkspaceParamKeys.Owners, true)) {
+              getWorkspaceOwners(workspaceContext.workspaceId.toString).map(_.map(_.value)).map(Option(_))
+            } else {
+              noFuture
+            }
+
+            def workspaceAuthorizationDomainFuture: () => Future[Option[Set[ManagedGroupRef]]] = () => if (getParam(options, WorkspaceAuthorizationDomain)) {
+              loadResourceAuthDomain(SamResourceTypeNames.workspace, workspaceContext.workspace.workspaceId, userInfo).map(Option(_))
+            } else {
+              noFuture
+            }
+
+            def workspaceSubmissionStatsFuture: () => slick.ReadAction[Option[WorkspaceSubmissionStats]] = () => if (getParam(options, WorkspaceParamKeys.WorkspaceSubmissionStats)) {
+              getWorkspaceSubmissionStats(workspaceContext, dataAccess).map(Option(_))
+            } else {
+              DBIO.from(noFuture)
+            }
```

Comment:
why are these all like
```
def workspaceAuthorizationDomainFuture: () => Future...
```
instead of
```
def workspaceAuthorizationDomainFuture(): Future...
```

Reply:
I wanted to make sure I wasn't launching the Futures too early, especially if I wasn't going to use their results. Looking at it again my `() => ...` approach may be overkill and I could simplify it